drivers/slipdev/slipdev.c

```diff
@@ -185,27 +185,28 @@ static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
         }
     }
     else {
-        int byte;
+        int byte = 0;
         bool escaped = false;
         uint8_t *ptr = buf;
 
         do {
             int tmp;
 
+            if ((unsigned)res == len) {
+                /* clear out unreceived packet */
+                while (byte != SLIPDEV_END) {
+                    byte = tsrb_get_one(&dev->inbuf);
+                }
+                return -ENOBUFS;
+            }
+
```

Comment:
```suggestion
            if (res >= len) {
                /* the result grew larger than the provided buffer
                clear out rest of the current packet, this package is lost */
                do {
                    byte = tsrb_get_one(&dev->inbuf);
                } while (byte != SLIPDEV_END);
                res = -ENOBUFS;
                break;
            }

```

Reply:
`do{ } while` to not depend on the initialization of `byte`
`res = .. ; break;` to avoid multiple returns
`(unsigned) res >= len` also catches negative `res` (for any reason) as bigger than `len`
and some comment cleanup (the old one sound like the package is unreceived (and might be still receivable)

Reply:
not sure if this helps with the ping issue but if not i don't think the original one helped either